qiskit_ibm_runtime/estimator.py

```diff
@@ -197,19 +197,35 @@ def __init__(
             skip_transpilation
         )
 
+        if isinstance(session, Session):
+            self._session = session
+        else:
+            backend = session or backend
+            self._session = get_default_session(service, backend)
+
+        backend_obj = self._session._service._backends.get(self._session._backend)
+
         if _options.optimization_level is None:
-            if _options.simulator and (
-                not hasattr(_options.simulator, "noise_model")
-                or asdict(_options.simulator)["noise_model"] is None
+            if (
+                backend_obj
+                and backend_obj.configuration().simulator
+                and (
+                    not hasattr(_options.simulator, "noise_model")
+                    or asdict(_options.simulator)["noise_model"] is None
+                )
```

Comment:
This logic really should be moved to `run()` because one can overwrite `noise_model` in `run()`. Also can have a utility function instead of repeating the same code.